packages/ove-asset-manager/README.md

```diff
@@ -1,7 +1,96 @@
+
+You are encouraged to run multiple copies of each Asset Processing Microservice, these should be registered with an instance of the Asset Manager and will then periodically check for Assets to process. Your database and Objectstore may each scale independently. 
+
+## Storage Model
+
+Assets will be stored on the object store on a one bucket per `Project` model. Every Asset is assigned a GUID on creation. This is used to identify it in the SQL database and on the object store. Within the bucket each Asset is uploaded to the root of a folder named by this ``GUID``. The filename of the asset will be preserved where possible. Asset processing services are free to place processed files and folders anywhere within the asset directory. 
+
+## Configuration
+
+Configuration of the service is achieved by modifying the ``Appsettings.json`` file. These settings are automatically over written by **Environmental Variables** (reflection of the json hierarcy should be acheived using ``__`` e.g. ``s3Client__Secret``). 
+
+### Configuring S3
+
+In common with all S3 compatible object stores the following three properties are required and should be set as follows in ``appsettings.json`` or overridden in environmental variables as discussed above.
+
+```  
+"s3Client": {
     "AccessKey": "secret",
     "Secret": "secret",
     "ServiceURL": "host"
-  },
\ No newline at end of file
+  }
+ ```
+ 
+### Configuring Maria DB
+The MariaDB configuration should be set as follows in `appsettings.json` or overridden in environmental variables as discussed above. 
+
+```
+  "MariaDB": { 
+    "ConnectionString": "Server=SERVER;Port=3306;Database=DATABASE;User=USERNAME;Password=PASSWORD;", 
+    "Version" :  "10.3.10"  
+  }
+```
+
+## Using the Asset Manager
+
+Assets should be uploaded to the asset manager either through the web interface or posting to the `/OVEAssetModelController/Create/` API. 
+Asset URL's on the objectstore may be found via the `/OVEAssetModelController/GetAssetURL?project=X&File=Y` API or by ID on  `/OVEAssetModelController/GetAssetURLbId?id=X`
+`/OVEAssetModelController/GetAssetURLbId?id=X`
+
+## Versioning
+
+The Asset Manager supports file versioning and will be default always provide the most recently uploaded/edited asset version when searching by Project and Asset Name. Previous versions may be found via the `/OVEAssetModelController/ListAssets/Project/` and `/OVEAssetModelController/ListAssets/Project/Name/` API's.
+
+## Processing state
+
+Asset processing services will update the ProcessingState of each asset using the ``/OVEAssetModelController/SetProcessingState/{id}/{state}/`` API. Assets in the unprocessed state (`0`) will be returned to processing microservices upon request. To reset the processing state of an asset and retry the processing you should use the ``/OVEAssetModelController/ResetProcessing/{id}`` API. 

```

Comment:
Hi James, the services are pull based,
if the processing state is reset then it will be reprocessed when a processing microservice requests for further work. I'll clarify this model earlier in the documentation.